File: Backend-new/app/schemas/champion.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional, Dict, Any

class ItemSimple(BaseModel):
    id: int
    name: str
    image: str
    model_config = ConfigDict(from_attributes=True)

class BuildResponse(BaseModel):
    item_1: Optional[ItemSimple]
    item_2: Optional[ItemSimple]
    item_3: Optional[ItemSimple]
    avg_placement: float
    win_rate: str
    model_config = ConfigDict(from_attributes=True)

class ItemStatResponse(BaseModel):
    item: Optional[ItemSimple]
    avg_placement: float
    win_rate: str
    pick_percent: str
    model_config = ConfigDict(from_attributes=True)

class TraitSimple(BaseModel):
    name: str
    slug: str
    model_config = ConfigDict(from_attributes=True)

class SkillResponse(BaseModel):
    name: str
    mana_start: int
    mana_max: int
    description: str
    ability_stats: Optional[Dict[str, Any]]
    icon_path: str
    model_config = ConfigDict(from_attributes=True)

class ChampionBase(BaseModel):
    id: int
    name: str
    slug: str
    cost: int
    rank: str
    avg_placement: float
    win_rate: str
    games_played: str
    pick_rate: str
    icon_path: str
    splash_path: str

class ChampionListResponse(ChampionBase):
    id: int
    skill: Optional[SkillResponse] = None
    traits: List[TraitSimple] = []
    best_items: List[ItemStatResponse] = []
    model_config = ConfigDict(from_attributes=True)

class ChampionDetailResponse(ChampionBase):
    base_stats: Optional[Dict[str, Any]]
    skill: Optional[SkillResponse]
    traits: List[TraitSimple] = []
    best_builds: List[BuildResponse] = []
    best_items: List[ItemStatResponse] = []
    model_config = ConfigDict(from_attributes=True)

File: Frontend/lib/types/champion.ts
export interface ApiSkill {
  id: number;
  name: string;
  description: string | null;
  icon_path: string | null;
}

export interface ApiChampion {
  id: number;
  name: string;
  cost: number;
  accent_color: string | null;
  icon_path: string | null;
  splash_path: string | null;
  skill: ApiSkill | null;
  avg_placement: number | null;
  top_4_rate: string | null;
  win_rate: string | null;
  match_count: number;
}

export interface ApiChampionSkill {
  name: string;
  mana_start: number;
  mana_max: number;
  description: string;
  ability_stats: Record<string, any>;
  icon_path: string;
}

export interface ApiChampionTrait {
  name: string;
  slug: string;
}

export interface ApiItemShort {
  id: number;
  name: string;
  image: string;
}

export interface ApiBestItem {
  item: ApiItemShort;
  avg_placement: number;
  win_rate: string;
  pick_percent: string;
}

export interface ApiBestBuild {
  item_1: ApiItemShort;
  item_2: ApiItemShort;
  item_3: ApiItemShort;
  avg_placement: number;
  win_rate: string;
}

export interface ApiChampionDetail {
  id: number;
  name: string;
  slug: string;
  cost: number;
  rank: string;
  avg_placement: number;
  win_rate: string;
  games_played: string;
  pick_rate: string;
  icon_path: string;
  splash_path: string;
  skill: ApiChampionSkill;
  traits: ApiChampionTrait[];
  base_stats: Record<string, string>; 
  best_builds: ApiBestBuild[];        
  best_items: ApiBestItem[];
}

